models/detr.py

```diff
@@ -58,11 +58,20 @@ def forward(self, samples: NestedTensor):
         """
         if isinstance(samples, (list, torch.Tensor)):
             samples = nested_tensor_from_tensor_list(samples)
-        features, pos = self.backbone(samples)
+
+        # embed image
+        features, pos        = self.backbone(samples)
+
+        # embed query
+        query_fts, query_pos = self.backbone(query)
+
+        # expend query to num_queries
+        query_fts = torch.cat(self.num_queries*[query_fts])
 
         src, mask = features[-1].decompose()
         assert mask is not None
-        hs = self.transformer(self.input_proj(src), mask, self.query_embed.weight, pos[-1])[0]
+        #hs = self.transformer(self.input_proj(src), mask, self.query_embed.weight, pos[-1])[0]
+        hs = self.transformer(self.input_proj(src), mask, query_fts, pos[-1])[0]
```

Comment:
Looking at this it probably makes more sense to add `query_fts` to `self.query_embed.weight` instead of just repeating `query_fts`.  I don't 100% understand what `query_embed` learns, but I think that's more reasonable.